libarchive/archive_write_set_format_pax.c

```diff
@@ -608,7 +608,15 @@ archive_write_pax_header(struct archive_write *a,
 	const time_t ustar_max_mtime = get_ustar_max_mtime();
 
 	/* Sanity check. */
+#if defined(_WIN32) && !defined(__CYGWIN__)
+	/* NOTE: If the caller supplied a pathname that fails WCS conversion (e.g.
+	 * if it is invalid UTF-8), we are expected to return ARCHIVE_WARN later on
+	 * in execution, hence the check for both pointers */
+	if ((archive_entry_pathname_w(entry_original) == NULL) &&
+	    (archive_entry_pathname(entry_original) == NULL)) {
```

Comment:
An alternative might be to introduce some `archive_entry_pathname_is_set`

Reply:
This is fine.